lib/r10k/mock_module.rb

```diff
@@ -0,0 +1,65 @@
+require 'puppet_forge'
+
+class R10K::MockModule
+
+  # @!attribute [r] title
+  #   @return [String] The forward slash separated owner and name of the module
+  attr_reader :title
+
+  # @!attribute [r] name
+  #   @return [String] The name of the module
+  attr_reader :name
+
+  # @param [r] dirname
+  #   @return [String] The name of the directory containing this module
+  attr_reader :dirname
+
+  # @!attribute [r] owner
+  #   @return [String, nil] The owner of the module if one is specified
+  attr_reader :owner
+
+  # @!attribute [r] path
+  #   @return [Pathname] The full path of the module
+  attr_reader :path
+
+  # @!attribute [r] version
+  #   @return [Pathname] The version, if it can be statically determined from args
+  attr_reader :version
+
+  # @param title [String]
+  # @param dirname [String]
+  # @param args [Array]
+  def initialize(title, dirname, args, environment=nil)
+    @title   = PuppetForge::V3.normalize_name(title)
+    @dirname = dirname
+    @owner, @name = parse_title(@title)
+    @path = Pathname.new(File.join(@dirname, @name))
+    @version = find_version(args)
```

Comment:
Do we need all these fields? Is there some hidden contract that requires them?

Reply:
We need the version and name in the current implementation (which requires the title). I thought we might need the path but I don't think so. This file was just a copy and pared down version of the module base class.